### ISSUE TITLE
azurerm_cosmosdb_account: Problem setting identity

### ISSUE THREAD

**Issue by mostrovoi** [hashicorp/terraform-provider-azurerm#14874]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_cosmosdb_account` version 2.91.0
* There is a problem setting identity for azurerm_cosmosdb_account, The only allowed value for identity is SystemAssigned with both S and A capital letters. However when running a plan the identity gets updated every time because of this:

 ~ identity {
          ~ type         = "systemassigned" -> "SystemAssigned"
            # (2 unchanged attributes hidden)
        }

So basically updates the type every time from non capital to capital letters. Is there any way to avoid the update of the identity type?

**Comment by mostrovoi** [comment]
Link to the documentation:

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cosmosdb_account

**Comment by aristosvo** [comment]
Hi @mostrovoi! Thanks for raising this issue.

Is this resource created with Terraform (or imported)? If created with Terraform, can you share the configuration for the `azurerm_cosmosdb_account`?

It shouldn't matter whether it is created or imported, but it would make it easier to figure out how this could happen in the first place and whether we should be able to reproduce it.

**Comment by tjanuario** [comment]
In my case, the resource was imported.  I am seeing the same behavior where systemassigned wants to be changed to SystemAssigned but the value for the identity in state remains as systemassigned after the apply which causes terraform to want to apply the change to SystemAssigned again.

**Comment by aristosvo** [comment]
@tjanuario Can you check what happens after changing it in state? Not something I'd advise to do in Production without having a backup, but it could solve the problem for now maybe.

After that you should check if it occurs again if you update a tag or some other setting to be sure it won't recur anytime you change one thing.

**Comment by tjanuario** [comment]
@aristosvo - I have never manually changed state before.  Do you mean just directly text editing that value?

**Comment by tjanuario** [comment]
@aristosvo - I pulled the state, updated "systemassigned" to "SystemAssigned" and changed serial to the next higher number.  I then pushed the state up and verified that the new state does have the correct "SystemAssigned" value.  However, when I run apply, I get the same results.

**Comment by tjanuario** [comment]
I did find a simple workaround and that was to ignore changes to the identity.type.  This way when the CosmosDB is created, the value SystemAssigned is accepted, and if it changes to systemassigned as it did in our case, the change is ignored and the configuration is valid.

lifecycle {
    ignore_changes = [
      identity[0].type
    ]
  }

**Comment by tjanuario** [comment]
Fix confirmed.  Thanks for the turnaround.
